src/agent/reqwest-retry/src/lib.rs

```diff
@@ -173,17 +181,34 @@ impl SendRetry for reqwest::RequestBuilder {
 mod test {
     use super::*;
 
-    // TODO: convert to feature-gated integration test.
-    #[ignore]
     #[tokio::test]
-    async fn empty_stack() -> Result<()> {
-        let resp = reqwest::Client::new()
-            .get("http://localhost:5000/api/testGet")
+    async fn retry_should_pass() -> Result<()> {
+        reqwest::Client::new()
+            .get("https://www.microsoft.com")
             .send_retry_default()
-            .await?;
-        println!("{:?}", resp);
+            .await?.error_for_status()?;
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn retry_should_fail() -> Result<()> {
+        let resp = reqwest::Client::new()
+            .get("http://localhost:81/test.txt")
+            .send_retry(
+                Duration::from_secs(1),
+                Duration::from_secs(3),
+                3i32,
+                to_backoff_response,
+            )
+            .await;
+
+        assert!(resp.is_err());
+        if let Err(err) = &resp {
+            let as_text = format!("{}", err);
+            assert!(as_text.contains("Maximum number of attempts reached for this request"));
+        }
```

Comment:
Instead of the preceding `assert!()`, maybe add an `else` branch here, which panics.